index.bs

```diff
@@ -4900,6 +5008,83 @@ as candidates to be employed in a [=registration ceremony=].
+
+In the opposite case, known as <dfn>all-sources credential properties</dfn>, it is possible for either the [=authenticator=] or the [=client=] to report on a certain [=credential property=]. Additionally, the [=client=] MAY so report if the [=client=] determines the [=authenticator=] failed to report, and the [=client=] possesses the appropriate data to report.
+
+At this time, one [=credential property=], which is an [=all-sources credential property=], is defined. It is known abstractly as the <dfn>resident key credential property</dfn>. It is a boolean value indicating whether or not the {{PublicKeyCredential}} returned as a result of a [=registration ceremony=] is a [=resident credential=] or a [=non-resident credential=].  
+
+: Extension identifier
+:: `credProps`
+
+: Operation applicability
+:: [=registration extension|Registration=]
+
+: Client extension input
+:: The Boolean value [TRUE] to indicate that this extension is requested by the [=[RP]=].
+    <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        boolean credProps;
+    };
+    </xmp>
+
+: Client extension processing
+::  1. Let |credPropsAuthenticatorInput| be the [=authenticator extension input=], created using the {{AuthenticationExtensionsClientInputs/credProps}} [=client extension input=].
+
+    1. When <a href='#CreateCred-InvokeAuthnrMakeCred'>invoking the authenticatorMakeCredential operation</a> include |credPropsAuthenticatorInput| in the |authenticatorExtensions| parameter.
+
+    1. If no [=authenticator=] indicates success, terminate this extension processing and do not return any outputs.
+
+    1. Assert: an [=authenticator=] [=CreateCred-AuthnrSuccess/indicates success=]. 
+
+    1. If <code>[=credentialCreationData/attestationObjectResult=].authData.[=authdataextensions|extensions=][`credProps`][`rk`]</code>:
+
+	<dl class="switch">
+            :  [=map-exists|Exists=]
+            ::  This is the [=authenticator=]'s result. Terminate these extension steps, and do not return any [=client extension output=].
+
+                Note: the [=authenticator=]'s result will be conveyed to the [=[RP]=] within the <code>[=credentialCreationData/attestationObjectResult=]</code>.
+
+            :  [=map-exists|Does not exist=]
+            :: See "Client extension output", below.
+        </dl>
+
+: Client extension output
+:: [=map-set|Set=] <code>[=credentialCreationData/clientExtensionResults=]["{{AuthenticationExtensionsClientOutputs/credProps}}"]</code> to the value of the |requireResidentKey| parameter that was used in the <a href='#CreateCred-InvokeAuthnrMakeCred'>invocation</a> of the [=authenticatorMakeCredential=] operation.
+
+    <xmp class="idl">
+    dictionary CredentialPropertiesOutput {
+        boolean rk;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        CredentialPropertiesOutput credProps;
+    };
+    </xmp>
+
+: Authenticator extension input
```

Comment:
Actually, why do we need authenticator extension at all.. I thought platform is going to figure out which one to create by inspecting authenticators and authenticators don't have to change.??

Reply:
I would remove Authenticator extension input/output sections for now. You already has "Client Extension Output" in your design. There is no need to add "Authenticator extension input/output here". Platform can figure out via other means. CTAP spec is the right place to figure out how best this information to be communicated back to the platform.

Reply:
All extensions are optional, so authenticators do not have to change. But the authenticator has the best knowledge of the credential it created, so why not allow the authenticator report that back if it can?

There's also the difference that authenticator extension outputs are signed, while client extension outputs are not.

Reply:
Removing authenticator extension component of the credProps extension per WG call on May 1.